scipy/stats/stats.py

```diff
@@ -6162,8 +6168,9 @@ def _permutation_ttest(a, b, permutations, axis=0, equal_var=True,
     mat = _broadcast_concatenate((a, b), axis=axis)
     mat = np.moveaxis(mat, axis, -1)
 
-    mat_perm, permutations = _data_permutations(mat, n=permutations,
-                                                random_state=random_state)
+    mat_perm, permutations = _data_partitions(mat, n=permutations, ma=na,
+                                              random_state=random_state)
+
```

Comment:
The docstring for the arg `permutations` should be updated. *the number of distinct permutations* → *the number of distinct partition*

Reply:
I considered that, but I didn't change it before because I thought it might confuse users. Part of the trouble is that "permutation" is not precisely the right word for this sort of thing. (After all, it led me to generate _all_ permutations of the data in my original implementation, when I should have done this.)

Do you think users will understand the use of "partition" when describing the "permutation" argument? I did adjust the notes to be more precise:

>When `permutations >= binom(n, k)`, an exact test is performed: the data are partitioned between the groups in each distinct way exactly once.

and I don't think the word "partition" would stand out as much there.

Reply:
Adjusted the wording. I think this works.

Reply:
RE: "the right word" 

I agree that `partition` and `permutation` can be a bit confusing-I'm not aware of a word to *describe* this quantity though-and I think the wording @mdhaber chose is pretty clear.